appium-tests/android/android.spec.js

```diff
@@ -0,0 +1,338 @@
+                .elementByXPath('//android.widget.TextView[@text="' + contactName.formatted + '"]')
+                .click()
+                .sleep(5000)
+                .context(webviewContext)
+                .elementById('info')
+                .getAttribute('innerHTML')
+                .then(function (html) {
+                    expect(html).toContain('"formatted": "' + contactName.formatted + '"');
+                })
+                .then(win, fail)
+                .finally(done);
+        }, 5 * MINUTE);
+
+        it('contacts.ui.spec.2 Update an existing contact', function (done) {
+            var oldContactName = contactsHelper.getContactName('Dooney', 'Evans');
+            var newContactName = contactsHelper.getContactName('Urist', 'McContact');
+            enterTest()
+                .execute(contactsHelper.getAddContactCode(oldContactName.formatted, oldContactName, "8 800 5553535"))
+                .sleep(5000)
+                .execute(contactsHelper.getRenameContactCode(oldContactName.formatted, newContactName.formatted, newContactName))
+                .sleep(5000)
+                .execute(contactsHelper.getPickContactCode())
+                .sleep(7000)
+                .context('NATIVE_APP')
+                .elementByXPath('//android.widget.TextView[@text="' + newContactName.formatted + '"]')
+                .click()
+                .sleep(5000)
+                .context(webviewContext)
+                .elementById('info')
+                .getAttribute('innerHTML')
+                .then(function (html) {
+                    expect(html).toContain('"formatted": "' + newContactName.formatted + '"');
+                })
+                .then(win, fail)
+                .finally(done);
+        }, 5 * MINUTE);
+
+        it('contacts.ui.spec.3 Create a contact with no info', function (done) {
+            var contactName = contactsHelper.getContactName();
+            enterTest()
+                .execute(contactsHelper.getAddContactCode())
+                .sleep(5000)
+                .execute(contactsHelper.getPickContactCode())
+                .sleep(7000)
+                .context('NATIVE_APP')
+                .elementByXPath('//android.widget.TextView[@text="(No name)"]')
+                .click()
+                .sleep(5000)
+                .context(webviewContext)
+                .elementById('info')
+                .getAttribute('innerHTML')
+                .then(function (html) {
+                    expect(html).toContain('"displayName": null');
+                })
+                .then(win, fail)
+                .finally(done);
+        }, 5* MINUTE);
+
+        it('contacts.ui.spec.4 Special characters and escape sequences', function (done) {
+            var contactName = contactsHelper.getContactName('Н€йромонах', 'ФеофаЊ');
+            enterTest()
+                .execute(contactsHelper.getAddContactCode(contactName.formatted, contactName, '4 5 6789'))
```

Comment:
I don't know if this string in particular occurs in other places, but please check if any literals like it can be factored out into constants.